src/cache/cache_las.c

```diff
@@ -1423,10 +1433,50 @@ __wt_find_lookaside_upd(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDAT
         upd->durable_ts = durable_timestamp;
         upd->start_ts = las_timestamp;
         upd->prepare_state = prepare_state;
-        /*
-         * Mark this update as external and to be discarded when not needed.
-         */
-        upd->ext = 1;
+
+        if (prepare_state == WT_PREPARE_INPROGRESS) {
+            WT_ASSERT(session, !modify);
+            WT_ASSERT(session, cbt->compare == 0);
+            upd_entry = NULL;
+            switch (page->type) {
+            case WT_PAGE_COL_FIX:
+            case WT_PAGE_COL_VAR:
+                WT_ASSERT(session, cbt->ins != NULL);
+                upd_entry = &cbt->ins->upd;
+                break;
+            case WT_PAGE_ROW_LEAF:
+                upd_entry =
+                  cbt->ins == NULL ? &page->modify->mod_row_update[cbt->slot] : &cbt->ins->upd;
+                break;
+            }
+
+            WT_ASSERT(session, upd_entry != NULL);
+            old_upd = *upd_entry;
+            upd_size = WT_UPDATE_MEMSIZE(upd);
+            cbt->modify_update = upd;
+            upd->next = old_upd;
+
+            /*
+             * Update serial will set the update pointer to null since it considers itself the owner
+             * of it now. However, we want to be able to keep a handle on the update so pass in a
+             * copy of the pointer.
+             */
+            tmp_upd = upd;
+            WT_ERR(__wt_update_serial(session, page, upd_entry, &tmp_upd, upd_size, false));
+
+            /*
+             * After reading in the prepared update, we need to delete it from lookaside. If it gets
+             * committed, the timestamp in the las key may differ so it's easier if we just get rid
+             * of it now and rewrite on eviction/commit/rollback.
+             */
+            WT_ERR(cursor->remove(cursor));
```

Comment:
We're in a pickle if we fail here since we've got the prepared update in memory and will rewrite on eviction and potentially duplicate the update with a different timestamp. I'm thinking this should maybe be a fatal error?

Reply:
I agree that failure in the remove is a rare scenario. Raising a fatal error could be good.

Reply:
Maybe sweep can detect such left-over prepared updates. If there is a committed update with the same or a newer timestamp than a prepared-update in lookaside, that prepare can be removed?

Reply:
If this ever happens, things have gone pretty far south. I wouldn't bother trying to keep going.

Reply:
3c5ff13

Panicking here.